exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/JoinPrel.java

```diff
@@ -158,6 +164,58 @@ protected void buildJoinConditions(List<JoinCondition> conditions,
     }
   }
 
+  private static boolean collapseExpandedIsNotDistinctFromExpr(RexCall call) {
```

Comment:
Nit: could you please create Calcite Jira to address it in the next version of Calcite?

Reply:
Created [CALCITE-3174] to address the issue.